rotkehlchen/exchanges/iconomi.py

```diff
@@ -60,13 +60,12 @@ def __init__(
         self.aust = A_AUST.resolve_to_asset_with_oracles()
 
     def _generate_signature(self, request_type: str, request_path: str, timestamp: str) -> str:
-        signed_data = f'{timestamp}{request_type.upper()}{request_path}'.encode()
-        signature = hmac.new(
-            self.secret,
-            signed_data,
-            hashlib.sha512,
+        signed_data = f'{timestamp}{request_type.upper()}{request_path}'
+        signature_digest = self.generate_hmac_signature(
+            message=signed_data,
+            digest_algorithm=hashlib.sha512,
         )
-        return base64.b64encode(signature.digest()).decode()
+        return base64.b64encode(bytes.fromhex(signature_digest)).decode()
```

Comment:
this one I beleive is similar to generate_hmac_sha256_b64_signature. Maybe it should take the algorithm as argument too